pkg/taskrunner/copy.go

```diff
@@ -0,0 +1,55 @@
+package taskrunner
+
+import (
+	"fmt"
+	"os"
+
+	cp "github.com/otiai10/copy"
+	"github.com/rs/zerolog/log"
+)
+
+type copyTask struct {
+	Source      string `json:"src"`
+	Destination string `json:"dest"`
+}
+
+func (e *copyTask) exec(c *Config) error {
+	l := log.With().
+		Str("task type", "copy").
+		Str("dest", e.Destination).
+		Logger()
+
+	// Convert source to list of files in case a glob is used
+	source, err := expandPath(c.Root, e.Source)
+	if err != nil {
+		return err
+	}
+
+	// Ensure output directory exists
+	if err := createDirIfNotExists(e.Destination, 0755); err != nil {
+		return err
+	}
+
+	for _, s := range source {
+		l.Debug().Str("src", s).Msg("Copying files")
+
+		if err := cp.Copy(s, e.Destination); err != nil {
+			return err
+		}
+
+	}
+
+	return nil
+}
+
+func createDirIfNotExists(dir string, perm os.FileMode) error {
+	if _, err := os.Stat(dir); !os.IsNotExist(err) {
+		return err
+	}
+
+	if err := os.MkdirAll(dir, perm); err != nil {
+		return fmt.Errorf("failed to create directory: '%s', error: '%s'", dir, err.Error())
```

Comment:
This one is a bit bad. While you did return the original error... You converted it into a string 😭 This is better all around:

```suggestion
		return fmt.Errorf("failed to create directory: '%s', error: %w", dir, err)
```

Additionally, you don't need to manually quote values. `Printf` already has a command for that, `%q`:

```suggestion
		return fmt.Errorf("failed to create directory: %q, error: %w", dir, err)
```